industrial_ci/src/tests/source_tests.sh

```diff
@@ -26,6 +26,27 @@ function install_catkin_lint {
     ici_install_pypi_pkgs_for_command catkin_lint "catkin-lint"
 }
 
+function clang_tidy_filter {
+    local source_dir=$1
+    local base_ref=$2
+    local -n _result=$3
+
+    pushd "$source_dir" > /dev/null || exit
+    if [ "$(git rev-parse --abbrev-ref HEAD)" == "$base_ref" ] ; then
+        _result=1
+    else
+        git fetch -q origin "$base_ref"
+        git branch -qf "$base_ref" FETCH_HEAD
```

Comment:
These are being done per-package.  They really only need to be done at the repo level.  I'm just not sure how to do that within this system.  The usage of this feature is really only intended to be used with one repo.  In testing though this doesn't seem to add measurable time to the runs this way (the fetch only takes time the first time it is called).